hyperspy/axes.py

```diff
@@ -655,6 +720,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(axes_manager=self)
+
+    def _on_view_changed(self):
+        self._update_attributes()
+        self.events.axes_changed.trigger()
```

Comment:
I think that changing the view has such a dramatic impact on anything that may connect to an `AxesManager` that having a separate event for it makes sense. I don't think that triggering an `axes_changed` event makes sense here. Yes, `slice` in an attribute of `DataAxis` but the axis itself didn't change, only the way we classify it. Triggering an `axes_changed` event in this case may cause problems down the line.

I agree that currently there is no consumer for a `view_changed` event. However, I think that it makes sense to add one as we'll soon put it to good use.